### PR TITLE
Add actions for copying data from PaaS to Azure

### PR DESCRIPTION
**Pull request by peteryates**
### Context

Prior to the move from GOV.UK PaaS to Azure we need to ensure our environments work properly with 'real' data. This PR introduces actions that copy it, environment by environment.

These actions will be reusable across environments, starting with staging. It uses the same process we used in ECF.

### CODE REVIEW

File: .github/actions/backup-paas-database/action.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        CF_SPACE_NAME: early-careers-framework-staging
+        INSTALL_CONDUIT: true
+
+    - name: Setup postgres client
+      uses: DFE-Digital/github-actions/install-postgres-client@master
+
+    - name: Backup database
+      shell: bash
+      run: |
+        cf conduit ecf-postgres-${{ inputs.environment }} -- pg_dump -E utf8 --clean --compress=1 --if-exists --no-owner --no-privileges --verbose -f backup-${{ inputs.environment }}.sql.gz
```
Comment (RMcVelia) [inline .github/actions/backup-paas-database/action.yml:39]: db is npq-registration-${{ inputs.environment }}-db

File: .github/actions/restore-paas-database/action.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Install kubectl
+      uses: azure/setup-kubectl@v3
+
+    - name: Install konduit
+      shell: bash
+      run: make install-konduit
+
+    - name: Restore database
+      shell: bash
+      run: bin/konduit.sh -i backup-${{ inputs.environment }}.sql.gz -c cpd-ecf-${{ inputs.environment }}-web -- psql
```
Comment (RMcVelia) [inline .github/actions/restore-paas-database/action.yml:43]: app is npq-registration-${{ inputs.environment }}-web

File: .github/actions/restore-paas-database/action.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    - name: Remove PaaS event triggers
+      shell: bash
+      run: |
+        bin/konduit.sh cpd-ecf-${{ inputs.environment }}-web -- psql -c 'drop event trigger forbid_ddl_reader; drop event trigger make_readable; drop event trigger reassign_owned;'
```
Comment (RMcVelia) [inline .github/actions/restore-paas-database/action.yml:48]: app is npq-registration-${{ inputs.environment }}-web

File: .github/workflows/copy-paas-database-to-azure.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          environment: staging
+          staging-username: ${{ secrets.GOVPAAS_STAG_USERNAME }}
+          staging-password: ${{ secrets.GOVPAAS_STAG_PASSWORD }}
+
+      - name: Restore
+        uses: ./.github/actions/restore-paas-database
+        with:
+          environment: staging
+          azure-credentials: ${{ secrets.AZURE_CREDENTIALS }}
```
Comment (RMcVelia) [inline .github/workflows/copy-paas-database-to-azure.yml:26]: need to add the secret

File: .github/actions/backup-paas-database/action.yml (hunk 1 above)
Comment (peteryates) [reply .github/actions/backup-paas-database/action.yml:39]: thank you, will fix. Was pulled onto something else and left this in flight 😩

**Review by RMcVelia — APPROVED** [review]

File: .github/workflows/copy-paas-database-to-azure.yml (hunk 4 above)
Comment (RMcVelia) [reply .github/workflows/copy-paas-database-to-azure.yml:26]: done

**Review by pmanrubia — APPROVED** [review]

**Pull request merged**